bitcoin/src/blockdata/block.rs

```diff
@@ -111,18 +113,40 @@ impl BlockHeader {
 #[derive(Copy, PartialEq, Eq, Clone, Debug, PartialOrd, Ord, Hash)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 #[cfg_attr(feature = "serde", serde(crate = "actual_serde"))]
-pub struct BlockVersion(pub i32);
+pub struct Version(i32);
```

Comment:
We're still keeping the inner value as `i32`?

Reply:
@nlanson the version is conceptually a signed integer. This is how it is represented in the reference implementation of Bitcoin.

Reply:
For casts, I see two casts to `u32` and one to `i32`, all to deal with the top-bit business. I don't think this is too bad.

I would ACK a PR which changed the internal repr to `u32` but I don't think it's necessary and I slightly prefer the current situation.

Reply:
`i32` is ridiculous, what does a negative version even mean? Time travel?

Unless Core somewhere actually uses something like -42 for version and doesn't treat it as a bag of bits I prefer `u32`. Thankfully newtype will shield the outside code from most of this nonsense.